packages/api-token-auth-client/src/ApiTokenAuthClient.ts

```diff
@@ -1,44 +1,60 @@
-import { AuthState, IAuthClient, SDKError, ERROR_CODES, PACKAGES } from '@8base/utils';
-
-type ApiTokenAuthClientOptions = {
+import {
+  IAuthState,
+  IAuthClient,
+  IStorage,
+  PACKAGES,
+  StorageAPI,
+  throwIfMissingRequiredParameters,
+} from '@8base/utils';
+
+interface IApiTokenAuthClientOptions {
   apiToken: string;
-};
+}
 
 /**
- * Create instacne of the api token auth client
- * @param {string} apiToken Api Token generated in 8base app.
+ * Creates instacne of the api token auth client
  */
 class ApiTokenAuthClient implements IAuthClient {
-  public apiToken: string;
-  public state: {};
+  private storageAPI: StorageAPI<IAuthState>;
+  private apiToken: string;
+
+  constructor(
+    options: IApiTokenAuthClientOptions,
```

Comment:
@vorobeez what do u think about using string instead of an object? I don't think we have any other options for the API token client